### DISCUSSION TITLE
Problems with create a room

### DISCUSSION THREAD

**Discussion by Romed78** [mynaparrot/plugNmeet-server#122] [Q&A]
When I try to create a new room I get the following error:
Fatal error: Declaration of Ramsey\Uuid\Uuid::unserialize(string $data): void must be compatible with Serializable::unserialize($serialized) in /var/home/.........../administrator/components/com_plugnmeet/helpers/libs/plugnmeet-sdk-php/vendor/ramsey/uuid/src/Uuid.php on line 307

does anyone know how to fix this error

Thanks

**Comment by jibon57** [comment]
What is your PHP version?

**Reply by Romed78** [reply]
PHP-Version „7.4.30“

**Reply by jibon57** [reply]
That problem shouldn't be in PHP 7.4. As far I know that may happen with PHP 8.1. Please double check your PHP version.

**Reply by Romed78** [reply]
php Version is already 7.4.30. Is the Problem joomla with the Version 3.10.11?

**Reply by jibon57** [reply]
No, that shouldn't be as we're using Joomla for our daily usage. The error message is indicating PHP related problem but I wasn't able to reproduce into my system. You may consider to update your PHP to 8.0.X

**Reply by Romed78** [reply]
The problem is that my platform with the modules doesn't work under php 8.0 at the moment (error ../plugins/system/less/lessc/less.php/Less.php on line 5305). So I can't update the page yet.

**Reply by jibon57** [reply]
I can give you a temporary solution. You can remove elements of this function: 
https://github.com/mynaparrot/plugNmeet-Joomla/blob/47e1562288f9edffe6c8e698968ca26e89d8a095/com_plugnmeet/admin/helpers/libs/plugnmeet-sdk-php/src/PlugNmeet.php#L245

And replace:

`return microtime();`

**Reply by jibon57** [reply]
For your case: /administrator/components/com_plugnmeet/helpers/libs/plugnmeet-sdk-php/src/PlugNmeet.php

**Reply by Romed78** [reply]
ist that right?
   {
        return microtime();
        return $uuid->toString();
    }

**Reply by jibon57** [reply]
No, keep only:
`return microtime();`
Remove other line

```
public function getUUID()
    {
        return microtime();
    }
```

**Reply by Romed78** [reply]
OK thanks. At the moment it looks like it will work. Are there any restrictions or misconduct as a result?

**Reply by jibon57** [reply]
No, it shouldn't be any problem. The previous code use to generate meeting id in UUID V4 format but that code didn't work in your system. Now your meeting id will be unix microsecond which is also valid. Meeting id only generate when you'll create your room from backend of Joomla. Once save meeting id will be store in database.

**Reply by Romed78** [reply]
Thank you very much. This is a great module for joomla. Thanks again.

**Reply by Romed78** [reply]
Sorry again, there's a new error:
invalid CreateRoomReq.RoomId: value does not match regex pattern "^[a-zA-Z0-9-_.:]+$"

**Reply by jibon57** [reply]
That's interesting 🤔. Can you change microtime() to `time()`?

**Reply by jibon57** [reply]
> Sorry again, there's a new error: invalid CreateRoomReq.RoomId: value does not match regex pattern "^[a-zA-Z0-9-_.:]+$"

You'll require to delete previous room or create new room from Joomla

**Reply by Romed78** [reply]
ok thanks is works :-)

**Reply by jibon57** [reply]
I've published new version which is v1.1.1. You can try that one. Above problem should be solved.

**Reply by Romed78** [reply]
Yes, it works great. Thanks
